src/sentry/static/sentry/app/views/settings/organizationDeveloperSettings/sentryApplicationDashboard/requestLog.tsx

```diff
@@ -0,0 +1,331 @@
+  }
+
+  renderBody() {
+    const {requests, eventType, errorsOnly, currentPage} = this.state;
+    const {app} = this.props;
+
+    const currentRequests = requests.slice(
+      currentPage * MAX_PER_PAGE,
+      (currentPage + 1) * MAX_PER_PAGE
+    );
+
+    return (
+      <React.Fragment>
+        <h4>{t('Request Log')}</h4>
+
+        <div>
+          <p>
+            {t(
+              'This log shows the status of any outgoing webhook requests from Sentry to your integration.'
+            )}
+          </p>
+
+          <RequestLogFilters>
+            <DropdownControl
+              label={eventType}
+              menuWidth="220px"
+              buttonProps={{style: {zIndex: theme.zIndex.header - 1}}}
+            >
+              {this.eventTypes.map(type => (
+                <DropdownItem
+                  key={type}
+                  onSelect={this.onChangeEventType}
+                  eventKey={type}
+                  isActive={eventType === type}
+                >
+                  {type}
+                </DropdownItem>
+              ))}
+            </DropdownControl>
+
+            <Button
+              onClick={this.onChangeErrorsOnly}
+              priority={errorsOnly ? 'primary' : 'default'}
+            >
+              <ErrorsOnlyCheckbox>
+                <Checkbox onChange={() => {}} checked={errorsOnly} />
+                {t('Errors Only')}
+              </ErrorsOnlyCheckbox>
+            </Button>
+          </RequestLogFilters>
+        </div>
+
+        <Panel>
+          <PanelHeader>
+            <TableLayout>
+              <div>{t('Time')}</div>
+              <div>{t('Status Code')}</div>
+              {app.status !== 'internal' && <div>{t('Organization')}</div>}
+              <div>{t('Event Type')}</div>
+              <div>{t('Webhook URL')}</div>
+            </TableLayout>
+          </PanelHeader>
+
+          {!this.state.loading ? (
+            <PanelBody>
+              {currentRequests.length > 0 ? (
+                currentRequests.map((request, idx) => (
+                  <PanelItem key={idx}>
+                    <TableLayout>
+                      <TimestampLink date={request.date} />
+                      <ResponseCode code={request.responseCode} />
+                      {app.status !== 'internal' && request.organization && (
+                        <div>{request.organization.name}</div>
+                      )}
+                      <div>{request.eventType}</div>
+                      <OverflowBox>{request.webhookUrl}</OverflowBox>
+                    </TableLayout>
+                  </PanelItem>
+                ))
+              ) : (
+                <EmptyMessage icon="icon-circle-exclamation">
+                  {t('No requests found in the last 30 days.')}
+                </EmptyMessage>
+              )}
+            </PanelBody>
+          ) : (
+            <LoadingIndicator />
+          )}
+        </Panel>
+
+        <PaginationButtons>
```

Comment:
Can we not use the `<Pagination>` component here?

Reply:
@billyvg Nope unfortunately because this isn't the same pagination using django models, the data is stored in Redis instead